src/parser/jobs/ast/modules/Draw.tsx

```diff
@@ -162,10 +174,10 @@ export default class Draw extends Module {
 		// Prepull consideration: + 1 play
 
 		// Begin Theoretical Max Plays calc
-		const fightDuration = this.parser.fight.end_time - this.parser.fight.start_time
-		const maxSleeveUses = Math.floor((fightDuration - (CARD_DURATION*2)) / (this.data.actions.SLEEVE_DRAW.cooldown * 1000)) + 1
+		const fightDuration = this.parser.pull.duration
+		const maxSleeveUses = Math.floor(Math.max(0, (fightDuration - (CARD_DURATION*2))) / (this.data.actions.SLEEVE_DRAW.cooldown * 1000)) + 1
```

Comment:
Well that was a very well written comment. I've struggled on this for awhile and it's been stuck at "good enough" until now. And with the new sleeve changes it's probably the best time to rewrite the whole algorithm to something that makes more sense.

1. Extra time after pressing SD is not much needed anymore since it only gives you one card, which can be swiftly delivered without having to weave a few times. I agree with you here, we don't need to account for that anymore.
2. Your point about this being a count of the max number of times the button is pressed, rather than a count of whether it's considered meaningful holds weight. It's important to make the distinction as to what this helps count. I believe in pulling data out first and making it visible, rather than skipping steps so people can see clearly the facts. Will make this change.
3. I'll look over your formula when I'm not bleary eyed and look into adapting it. 

Thank you so much for taking the time to look through all that spaghetti

Reply:
Thank you! I was worried there was going to be too much rambling, since I was writing it at about 4 in the morning. :sweat_smile: 

1. The function I wrote scales for multiple cards pre-5.3, and in 5.3 looks for one GCD for Sleeve Draw and another for Play, so that's probably close enough? I recognize these aren't actually GCD skills and could maybe be played faster, but between trying to ensure optimal card plays and weaving Draw > Play, I think one GCD per card works as a baseline? Feel free to disagree, I would like to hear your thoughts - I could very easily be over/underestimating how long it takes to identify and play a card, I'm usually too busy trying to actually play the card to to track how long it usually takes me.
    
    If we don't want to require a GCD for Play at all starting in 5.3, we could adapt the function to return a list of possible activations and the number of cards there was time to play during that activation. If we did that:
    - Before 5.3 we would just have to sum the card counts from the results
    - Starting in 5.3, we could pass in `0` for `cardsToPlay`, making the function only check for enough time to activate the ability. Then we could use the length of the list of activations as the number of possible plays, since each activation would equal 1 Play.
2. Thanks, I'm glad my confusion helped. :smile: Can you clarify for me if you're saying the number you want to start reporting is the "physical button press" maximum, or the "enough time to be effective" maximum? It sounds like the physical maximum, but just double-checking.
3. I tested the function locally as I was writing it and it seemed to work for 5.3 parse, so hopefully there won't be much adaptation required.